ParquetEncoding/EncodingRead.go

```diff
@@ -205,16 +205,21 @@ func ReadBitPacked(bytesReader *bytes.Reader, header uint64, bitWidth uint64) ([
 		return res, err
 	}
 	bytesBuf := make([]byte, byteCnt)
-	if _, err = bytesReader.Read(bytesBuf); err != nil {
-		return res, err
+	if byteCnt > 0 {
+		if _, err = bytesReader.Read(bytesBuf); err != nil {
+			return res, err
+		}
 	}
 
 	i := 0
 	var resCur uint64 = 0
 	var resCurNeedBits uint64 = bitWidth
 	var used uint64 = 0
 	var left uint64 = 8 - used
-	b := bytesBuf[i]
+	var b uint8 = 0
+	if i < len(bytesBuf) {
+		b = bytesBuf[i]
```

Comment:
could you just move this inside the for loop instead?

```
for i < len(bytesBuf) {
    b := bytesBuf[i]
   ...
```

Reply:
No, b is not updated for every iteration